osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/GetKeysService.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2021 Alliander N.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ */
+package org.opensmartgridplatform.adapter.protocol.dlms.application.services;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.DlmsDevice;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.SecurityKeyType;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.GetKeysRequestDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.GetKeysResponseDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.KeyDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.SecretTypeDto;
+import org.opensmartgridplatform.shared.security.RsaEncrypter;
+import org.opensmartgridplatform.ws.schema.core.secret.management.SecretType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GetKeysService {
+
+  private final SecretManagementService secretManagementService;
+  private final RsaEncrypter rsaEncrypter;
```

Comment:
Would another name be an improvement, that explains more about the particular encryption that is applied by this `RsaEncrypter` to make it clearer what this encrypter is for, not only that is applies RSA encryption? (Something like `gxfResponseMessageSecretEncrypter` comes to mind, but I can imagine you could possibly come up with something even clearer.)

Reply:
I renamed it to `keyEncrypter`. Adding `responseMessage` to the name implies that the service is aware that the keys end up in a response message from gxf, but it isn't.